crates/asm/subprotocols/bridge-v1/src/state/bridge.rs

```diff
@@ -0,0 +1,897 @@
+    /// The withdrawal processing information is returned to the caller for storage in MohoState
+    /// and later use by Bridge proof.
+    ///
+    /// # Parameters
+    ///
+    /// - `tx` - The withdrawal fulfillment transaction
+    /// - `withdrawal_info` - Parsed withdrawal information containing operator, deposit details,
+    ///   and withdrawal amounts
+    ///
+    /// # Returns
+    ///
+    /// - `Ok(WithdrawalProcessedInfo)` - The processed withdrawal information if transaction passes
+    ///   validation
+    /// - `Err(WithdrawalValidationError)` - If validation fails for any reason
+    ///
+    /// # Errors
+    ///
+    /// Returns error if:
+    /// - No assignment exists for the referenced deposit
+    /// - The operator doesn't match the assigned operator
+    /// - The withdrawal specifications don't match the assignment
+    /// - The deposit referenced in the withdrawal doesn't exist
+    pub fn process_withdrawal_fulfillment_tx(
+        &mut self,
+        tx: &Transaction,
+        withdrawal_info: &WithdrawalFulfillmentInfo,
+    ) -> Result<OperatorClaimUnlock, WithdrawalValidationError> {
+        self.validate_withdrawal(withdrawal_info)?;
+
+        // Remove the assignment from the table to mark withdrawal as fulfilled
+        // Safe to unwrap since validate_withdrawal ensures the assignment exists
+        let removed_assignment = self
+            .assignments
+            .remove_assignment(withdrawal_info.deposit_idx)
+            .expect("Assignment must exist after successful validation");
+
+        Ok(OperatorClaimUnlock {
+            withdrawal_txid: tx.compute_txid().into(),
+            deposit_txid: removed_assignment.deposit_txid(),
+            deposit_idx: removed_assignment.deposit_idx(),
+            operator_idx: withdrawal_info.operator_idx,
+        })
+    }
+
+    /// Validates the parsed withdrawal it against assignment information.
+    ///
+    /// This function takes already parsed withdrawal information and validates it
+    /// against the corresponding assignment entry. It checks that:
+    /// - An assignment exists for the withdrawal's deposit
+    /// - The operator performing the withdrawal matches the assigned operator
+    /// - The withdrawal amounts and destinations match the assignment specifications
+    ///
+    /// # Parameters
+    ///
+    /// - `withdrawal_info` - Parsed withdrawal information containing operator, deposit details,
+    ///   and amounts
+    ///
+    /// # Returns
+    ///
+    /// - `Ok(())` - If the withdrawal is valid according to assignment information
+    /// - `Err(WithdrawalValidationError)` - If validation fails for any reason
+    ///
+    /// # Errors
+    ///
+    /// Returns error if:
+    /// - No assignment exists for the referenced deposit
+    /// - The operator doesn't match the assigned operator
+    /// - The withdrawal specifications don't match the assignment
+    fn validate_withdrawal(
```

Comment:
This is the bridge withdrawal after a fulfillment (and claim), right? why do we even needs this logic on the OL?